### PR TITLE
FINERACT-1905: New repayment strategy

### PR DESCRIPTION
**Pull request by adamsaghy**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanChargeEffectiveDueDateComparator.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package org.apache.fineract.portfolio.loanaccount.domain;
+
+import java.util.Comparator;
+
+/**
+ * Sort loan charges by effective due date
+ *
+ * Null values go to the end
+ */
+public class LoanChargeEffectiveDueDateComparator implements Comparator<LoanCharge> {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanChargeEffectiveDueDateComparator.java]: Make this a singleton class if there's a specific instance to use (i.e. drop the public constructor)

Comment (adamsaghy) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanChargeEffectiveDueDateComparator.java]: 👍

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/starter/LoanAccountAutoStarter.java [hunk 2]
```diff
@@ -78,6 +79,12 @@ public RBILoanRepaymentScheduleTransactionProcessor rbiLoanRepaymentScheduleTran
         return new RBILoanRepaymentScheduleTransactionProcessor();
     }
 
+    @Bean
+    @ConditionalOnProperty("fineract.loan.transactionprocessor.due-date-respective.enabled")
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/starter/LoanAccountAutoStarter.java]: Would be better to have the property mapped to the FineractProperties class and then using a PropertiesCondition for this. That way it's going to be type-safe and the property name is not duplicated.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/config/FineractProperties.java [hunk 3]
```diff
@@ -295,4 +297,33 @@ public static class UserNotificationSystemProperties {
 
         private boolean enabled;
     }
+
+    @Getter
+    @Setter
+    public static class FineractLoanProperties {
+
+        private FineractTransactionProcessorProperties transactionprocessor;
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/config/FineractProperties.java]: Java naming convention pls. Should be `transactionProcessor`, note the capital `P`

**Review by galovics — CHANGES_REQUESTED** [review]

**Review by ruchiD — COMMENTED** [review]
LGTM

**Review by galovics — APPROVED** [review]

**Pull request merged**